src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3485,7 +3497,7 @@ static bool
 AggregateEnabledCustom(Aggref *aggregateExpression)
 {
 	if (aggregateExpression->aggorder != NIL ||
-		list_length(aggregateExpression->args) != 1)
+		list_length(aggregateExpression->args) == 0)
```

Comment:
Can we ever have aggregateExpression with zero args?  My understanding that it is not allowed by Postgres (as of now).

I think it is better to have this check anyway. Might be nice to have a single line comment that we have this check for being defensive?

Reply:
I think we could have that case. From "CREATE AGGREGATE" documentation: to create a zero-argument aggregate function, write * in place of the list of argument specifications. (An example of such an aggregate is count(*).)